src/mono_hash_map.rs

```diff
@@ -18,8 +18,11 @@ use crate::AllocMap;
 pub struct MonoHashMap<K: Hash + Eq, V>(RefCell<FxHashMap<K, Box<V>>>);
 
 impl<K: Hash + Eq, V> MonoHashMap<K, V> {
-    pub fn values<T>(&self, f: impl FnOnce(&mut dyn Iterator<Item=&V>) -> T) -> T {
-        f(&mut self.0.borrow().values().map(|v| &**v))
+    /// This function exists for priroda to be able to iterate over all evaluator memory
+    ///
+    /// The memory of constants does not show up in this list.
+    pub fn iter<T>(&self, f: impl FnOnce(&mut dyn Iterator<Item=(&K, &V)>) -> T) -> T {
+        f(&mut self.0.borrow().iter().map(|(k, v)| (k, &**v)))
```

Comment:
I always wondered, why is the signature so strange? Why not a `fn iter(...) -> impl Iterator<Item=...>`?

Reply:
I don't think that can work. We'd need a way to `DerefMut` the result of `std::cell::Ref::map`. While that can be made sound, the libstd just isn't there.

Reply:
Ah dang, `next()` needs `&mut`... we'd not want `Ref::map`, we'd want to return a pair of a `Ref` and a ptr into it, which is not possible in safe code.

Please add a comment explaining that this is closure-based because the `RefCell` needs to be borrowed for the entire duration of the iteration, which doesn't really work with `-> impl Iterator`.